packages/sit-onyx/src/i18n/index.spec.ts

```diff
@@ -0,0 +1,165 @@
+  i18n = injectI18n();
+
+  expect(i18n.locale.value).toBe("a");
+  locale.value = "b";
+  expect(i18n.locale.value).toBe("b");
+});
+
+test("should translate with/without placeholders", () => {
+  provideI18n({
+    locale: "en-US",
+    messages: {
+      "en-US": {
+        plain: "Hello World",
+        placeholder: "Hello {firstName} {lastName}",
+      },
+    },
+  });
+
+  const { t } = injectI18n();
+
+  let message = t.value("plain" as TestTranslationKey);
+  expect(message).toBe("Hello World");
+
+  message = t.value("placeholder" as TestTranslationKey, {
+    firstName: "John",
+    lastName: "Doe",
+  });
+  expect(message).toBe("Hello John Doe");
+
+  // should return empty string for missing translation
+  message = t.value("does.not.exist" as TestTranslationKey);
+  expect(message).toBe("");
+});
+
+test("should translate with pluralization", () => {
+  provideI18n({
+    locale: "en-US",
+    messages: {
+      "en-US": {
+        pluralization: "Zero items | 1 item | {n} items",
+        pluralizationWithoutZero: "1 item | {n} items",
+        withoutPluralization: "{n} items",
+      },
+    },
+  });
+
+  const { t } = injectI18n();
+
+  let message = t.value("pluralization" as TestTranslationKey, { n: 0 });
+  expect(message).toBe("Zero items");
+
+  message = t.value("pluralization" as TestTranslationKey, { n: 1 });
+  expect(message).toBe("1 item");
+
+  message = t.value("pluralization" as TestTranslationKey, { n: 2 });
+  expect(message).toBe("2 items");
+
+  message = t.value("pluralization" as TestTranslationKey, { n: -1 });
+  expect(message).toBe("-1 items");
+
+  message = t.value("pluralization" as TestTranslationKey);
+  expect(message).toBe("1 item");
+
+  message = t.value("pluralizationWithoutZero" as TestTranslationKey, { n: 0 });
+  expect(message).toBe("0 items");
+
+  message = t.value("pluralizationWithoutZero" as TestTranslationKey, { n: 1 });
+  expect(message).toBe("1 item");
+
+  message = t.value("pluralizationWithoutZero" as TestTranslationKey, { n: 2 });
+  expect(message).toBe("2 items");
+
+  message = t.value("withoutPluralization" as TestTranslationKey, { n: 0 });
+  expect(message).toBe("0 items");
+
+  message = t.value("withoutPluralization" as TestTranslationKey, { n: 1 });
+  expect(message).toBe("1 items");
+
+  message = t.value("withoutPluralization" as TestTranslationKey, { n: 2 });
+  expect(message).toBe("2 items");
+});
+
+test("should update translation when locale changes", () => {
+  const locale = vue.ref("en-US");
+  provideI18n({
+    locale,
+    messages: {
+      "en-US": {
+        helloWorld: "Hello World",
+      },
+      "de-DE": {
+        helloWorld: "Hallo Welt",
+      },
+    },
+  });
+  const { t } = injectI18n();
+
+  let message = t.value("helloWorld" as TestTranslationKey);
+  expect(message).toBe("Hello World");
+
+  locale.value = "de-DE";
+
+  message = t.value("helloWorld" as TestTranslationKey);
+  expect(message).toBe("Hallo Welt");
```

Comment:
wouldn't something like this reflect the expectation/real-world-scenario better?

```suggestion
  const message = vue.computed(()=> t.value("helloWorld" as TestTranslationKey));

  expect(message.value).toBe("Hello World");
  locale.value = "de-DE";
  expect(message.value).toBe("Hallo Welt");
```

Reply:
Yes, like it :)